packages/ods/src/components/combobox/tests/rendering/ods-combobox.e2e.ts

```diff
@@ -269,6 +269,176 @@ describe('ods-combobox rendering', () => {
         expect(await isGroupVisible('group2')).toBe(true);
       });
     });
+    describe('onGroupsSlotChange', () => {
+      it('should correctly update groups when adding a group asynchronously', async() => {
+        async function isGroupVisible(id: string): Promise<boolean> {
+          return page.evaluate((elementId) => {
+            const groupTitle = document.querySelector(`#${elementId}`)?.shadowRoot?.querySelector('.ods-combobox-group__title');
+            return !!groupTitle && !groupTitle.classList.contains('ods-combobox-group__title--hidden');
+          }, id);
+        }
+
+        await setup(`
+          <ods-combobox>
+            <ods-combobox-group id="group1">
+              <span slot="title">Group 1</span>
+              <ods-combobox-item value="dummy1">Dummy 1</ods-combobox-item>
+            </ods-combobox-group>
+        `);
+
+        await (await page.find('ods-combobox >>> ods-input')).click();
+        await page.waitForChanges();
+
+        expect(await isOpen()).toBe(true);
+        expect(await isGroupVisible('group1')).toBe(true);
+
+        setTimeout(async() => {
+          await page.evaluate(() => {
+          document.querySelector('ods-combobox')!.innerHTML = `
+            <ods-combobox-group id="group1">
+              <span slot="title">Group 1</span>
+              <ods-combobox-item value="dummy1">Dummy 1</ods-combobox-item>
+            </ods-combobox-group>
+            <ods-combobox-group id="group2">
+              <span slot="title">Group 2</span>
+              <ods-combobox-item value="dummy2">Dummy 2</ods-combobox-item>
+            </ods-combobox-group>
+          `;
+          });
+        }, 1000);
```

Comment:
we can't rely on timeout as it will fails randomly on slow CI.
code should run on more robust call (the waitForFunction should be enough, if not we need to investigate why and how to fix)